compiler/src/dotty/tools/dotc/transform/RenameLifted.scala

```diff
@@ -9,14 +9,15 @@ import dotty.tools.dotc.core.Names._
 import dotty.tools.dotc.core.Phases
 import dotty.tools.dotc.core.SymDenotations.SymDenotation
 import dotty.tools.dotc.core.Symbols._
-import dotty.tools.dotc.transform.TreeTransforms.MiniPhaseTransform
+import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
 
 /** Renames lifted classes to local numbering scheme */
 class RenameLifted extends MiniPhaseTransform with SymTransformer { thisTransformer =>
 
   override def phaseName = "renameLifted"
 
-  override def runsAfterGroupsOf: Set[Class[_ <: Phases.Phase]] = Set(classOf[RestoreScopes])
+  // Not clear why this should run after restoreScopes
+  // override def runsAfterGroupsOf: Set[Class[_ <: Phases.Phase]] = Set(classOf[RestoreScopes])
```

Comment:
@nicolasstucki Can you comment on that ?

Reply:
Originally it was on `Flatten` because this transformation acts on the flattened names. But @DarkDimius noted that the flattening transformation finishes in `RestoreScopes`.